tests/test_validation.py

```diff
@@ -0,0 +1,184 @@
+def test_report(context, app):
+    app.authorize([
+        'spinta_report_insert',
+        'spinta_report_getone',
+    ])
+
+    resp = app.post('/report', json={
+        'type': 'report',
+        'report_type': 'simple',
+        'status': 'valid',
+        'count': '42',
+        'valid_from_date': '2019-04-20',
+        'update_time': '2019-04-20 03:14:15',
+        'notes': [{
+            'note': 'hello report',
+            'note_type': 'test',
+            'create_date': '2019-04-20',
+        }]
+    })
+    assert resp.status_code == 201
+
+    data = resp.json()
+    id = data['id']
+    resp = app.get(f'/report/{id}')
+    assert resp.status_code == 200
+
+    # FIXME: should return date/datetime strings instead of unix timestamps?
+    assert resp.json() == {
+        'id': id,
+        'type': 'report',
+        'report_type': 'simple',
+        'status': 'valid',
+        'count': 42,
+        'valid_from_date': 1555718400,
+        'update_time': 1555730055,
+        'notes': [{
+            'note': 'hello report',
+            'note_type': 'test',
+            'create_date': 1555718400,
+        }]
+    }
+
+
+def test_invalid_report_int(context, app):
+    app.authorize([
+        'spinta_report_insert',
+    ])
+
+    resp = app.post('/report', json={
+        'type': 'report',
+        'report_type': 'simple',
+        'status': 'valid',
+        'count': 'c0unt',  # invalid conversion to int
+    })
+
+    assert resp.status_code == 400
+    assert resp.json() == {
+        "error": "invalid literal for int() with base 10: 'c0unt'",
```

Comment:
Probably this is for another pull request, but here I would like to see field name, otherwise these error messages could be really difficult to understand and fix.

Reply:
This is indeed for later.